### PR TITLE
Add ReZero and ScaleNorm support

### PR DESCRIPTION
**Pull request by tomweingarten**
Also engaging in some poor PR hygiene by fixing a simple bug with the ff_activation parameter.

### CODE REVIEW

**Comment by lucidrains** [comment]
lgtm!

**Comment by lucidrains** [comment]
@tomweingarten what configuration have you had the most luck with? rezero or scalenorm?

**Comment by tomweingarten** [comment]
@lucidrains I haven't run any long studies yet but my initial results show ScaleNorm to be faster to converge

I haven't seen any cases with either diverging. The Adafactor optimizer seems to work very well at keeping both stable.

**Comment by lucidrains** [comment]
@tomweingarten Yes, I believe I have noticed the same, last night, it came to me there is a connection between scale norm and https://arxiv.org/abs/2003.07845 , where they relax the zero-meaning

**Comment by lucidrains** [comment]
@tomweingarten very interesting for Rezero! I have noticed divergence on bigger datasets (common crawl), but I shall try it again given your testimony and see if some more aggressive gradient clipping can fix that

**Comment by tomweingarten** [comment]
Looking forward to hear how it works for you! I'd also recommend either A)
using an optimizer with variable learning rates like Adafactor or B) using
a separate learning rate for the residual weights. Otherwise even with
gradient clipping you can see divergence caused by the momentum over
multiple steps.

**Comment by lucidrains** [comment]
@tomweingarten yes, you did allude to this different learning rate in some footnote in the rezero paper, i'll reread it tonight. thanks!
